File: client/app/src/main/java/it/unito/ium/myreps/model/services/api/ServerError.java
package it.unito.ium.myreps.model.services.api;

import org.jetbrains.annotations.NotNull;

public enum ServerError {
    LOGIN_DISABLED("Login for this user is disabled"),
    LOGIN_FAILED("Wrong email or password"),
    USER_ALREADY_EXIST("This user is already registered"),
    USER_NOT_FOUND("User not found"),
    BOOKING_NOT_FOUND("Booking not found"),
    LESSON_NOT_FOUND("Lesson not found"),
    COURSE_NOT_FOUND("Course not found"),
    BAD_REQUEST("Bad request format"),
    UNAUTHORIZED("You are not authorized"),
    NOT_FOUND("Endpoint not found"),
    METHOD_NOT_ALLOWED("Method not allowed"),
    SERVER_ERROR("Server error"),
    UNKNOWN_ERROR("Unknown error"),
    SERVER_OFFLINE("Server seems offline");

    // ---------------------------------------------------------------

    private final String humanReadable;

    ServerError(String humanReadable) {
        this.humanReadable = humanReadable;
    }

    public static ServerError fromString(String s) {
        ServerError serverError = null;
        try {
            serverError = ServerError.valueOf(s.toUpperCase());
        } catch (EnumConstantNotPresentException ignored) {
        }
        return serverError;
    }

    @NotNull
    @Override
    public String toString() {
        return humanReadable;
    }
}

File: server/server_error.py
#!/usr/bin/python3
from flask import make_response


def server_error(s: str):
    s = s.upper()
    string = server_error.__data.get(s, None)
    if string is None:
        string = server_error.__data.get('UNKNOWN_ERROR')
        return make_response({'ok': False, 'error': 'UNKNOWN_ERROR'}, string)
    return make_response({'ok': False, 'error': s}, string)


server_error.__data = {
    'USER_ALREADY_EXIST': 400,
    'BAD_REQUEST': 400,
    'UNAUTHORIZED': 401,
    'AUTH_FAILED': 401,
    'LOGIN_FAILED': 401,
    'NOT_FOUND': 404,
    'USER_NOT_FOUND': 404,
    'BOOKING_NOT_FOUND': 404,
    'LESSON_NOT_FOUND': 404,
    'COURSE_NOT_FOUND': 404,
    'METHOD_NOT_ALLOWED': 405,
    'LOGIN_DISABLED': 500,
    'SERVER_ERROR': 500,
    'UNKNOWN_ERROR': 204,
    'SERVER_OFFLINE': -1
}
